core/llm.go

```diff
@@ -634,11 +651,10 @@ func (llm *LLM) Sync(ctx context.Context, dag *dagql.Server) error {
 	if err := llm.allowed(ctx); err != nil {
 		return err
 	}
-	var err error
 	llm.once.Do(func() {
-		err = llm.loop(ctx, dag)
+		llm.err = llm.loop(ctx, dag)
 	})
-	return err
+	return llm.err
```

Comment:
this is a bugfix, isn't it? is there a test? (not requesting one rn, just noting for posterity that there's a case here where you can lose the error by calling sync twice

Reply:
Yeah, there's a missing test here, and I don't think there's a very obvious place to put one at the moment. :confused: